src/generator/AutoRest.Go/Model/MethodGo.cs

```diff
@@ -245,12 +245,16 @@ public List<string> PrepareDecorators
               );
         }
 
-        if (RequestHeaders.Any())
+        if (HeaderParameters.Any())
```

Comment:
It looks like `RequestHeaders` is used for the modeler implementation. `HeaderParameters` comes from the already built model.